clustering/jgroups/extension/src/main/java/org/jboss/as/clustering/jgroups/subsystem/JGroupsSubsystemXMLReader.java

```diff
@@ -372,38 +329,44 @@ private void parseTransport(XMLExtendedStreamReader reader, PathAddress stackAdd
                     break;
                 }
                 case DEFAULT_EXECUTOR: {
-                    readAttribute(reader, i, operation, TransportResourceDefinition.ThreadingAttribute.DEFAULT_EXECUTOR);
+                    if (this.schema.since(JGroupsSchema.VERSION_9_0)) {
+                        throw ParseUtils.unexpectedAttribute(reader, i);
+                    }
+                    JGroupsLogger.ROOT_LOGGER.attributeIgnored(attribute.getLocalName(), reader.getLocalName());
                     break;
                 }
                 case OOB_EXECUTOR: {
-                    readAttribute(reader, i, operation, TransportResourceDefinition.ThreadingAttribute.OOB_EXECUTOR);
+                    if (this.schema.since(JGroupsSchema.VERSION_9_0)) {
+                        throw ParseUtils.unexpectedAttribute(reader, i);
+                    }
+                    JGroupsLogger.ROOT_LOGGER.attributeIgnored(attribute.getLocalName(), reader.getLocalName());
                     break;
                 }
                 case TIMER_EXECUTOR: {
-                    readAttribute(reader, i, operation, TransportResourceDefinition.ThreadingAttribute.TIMER_EXECUTOR);
+                    if (this.schema.since(JGroupsSchema.VERSION_9_0)) {
+                        throw ParseUtils.unexpectedAttribute(reader, i);
+                    }
+                    JGroupsLogger.ROOT_LOGGER.attributeIgnored(attribute.getLocalName(), reader.getLocalName());
                     break;
                 }
                 case THREAD_FACTORY: {
-                    readAttribute(reader, i, operation, TransportResourceDefinition.ThreadingAttribute.THREAD_FACTORY);
+                    if (this.schema.since(JGroupsSchema.VERSION_9_0)) {
+                        throw ParseUtils.unexpectedAttribute(reader, i);
+                    }
+                    JGroupsLogger.ROOT_LOGGER.attributeIgnored(attribute.getLocalName(), reader.getLocalName());
```

Comment:
These branches are identical, I would prefer a fallthrough here.

Reply:
No.  Falling though would preclude the addition of any new attributes.

Reply:
@pferraro Hm, can you explain how?

```
                case DEFAULT_EXECUTOR:
                case THREAD_FACTORY:
                case TIMER_EXECUTOR:
                case OOB_EXECUTOR: {
                    if (this.schema.since(JGroupsSchema.VERSION_9_0)) {
                        throw ParseUtils.unexpectedAttribute(reader, i);
                    }
                    JGroupsLogger.ROOT_LOGGER.attributeIgnored(attribute.getLocalName(), reader.getLocalName());
                    break;
                }
```

Reply:
I thought you meant to fall through to the default.  This is fine.